modules/ROOT/pages/authentication-authorization/limitations.adoc

```diff
@@ -332,3 +332,51 @@ Due to this, each node with the `:Person` label needs to be accessed and examine
 Due to the additional data accesses that the security checks need to do, this operation will be slower compared to executing the query as an unrestricted user.
 
 |===
+[[property-based-access-control-limitations]]
+=== Property-based access control limitations
+The same goes for when adding security rules based on property values. Similar to above we have a `restricted` and a `free` roles as an example:
+
+[source, cypher]
+----
+GRANT TRAVERSE ON GRAPH * FOR (n:Customer) WHERE n.secret <> true TO restricted;
+GRANT TRAVERSE ON GRAPH * ELEMENTS * TO free;
+----
+
+When executing query:
+
+[source, cypher]
+----
+MATCH (n:Customer)
+RETURN count(n)
+----
+
+For both roles the execution plan will look like this:
+
+----
++--------------------------+
+| Operator                 |
++--------------------------+
+| +ProduceResults          |
+| |                        +
+| +NodeCountFromCountStore |
++--------------------------+
+----
+
+Internally, however, very different operations need to be executed.
+The following table illustrates the difference:
+
+[%header,cols=2*]
+|===
+|User with `free` role
+|User with `restricted` role
+
+|The database can access the count store and retrieve the total number of nodes with the label `:Customer`.
+
+This is a very quick operation.
+
+|The database cannot access the count store because it must make sure that only traversable nodes with the desired label `:Customer` are counted.
+Due to this, each node with the `:Customer` label needs to be accessed and examined to make sure that they do not have property `secret` set to `true`.
+
+Due to the additional data accesses that the security checks need to do, this operation will be slower compared to executing the query as an unrestricted user.
+
+|===
```

Comment:
```suggestion
MATCH (n:Customer)
RETURN n
----

For both roles the execution plan will look like this:

----
+--------------------------+
| Operator                 |
+--------------------------+
| +ProduceResults          |
| |                        +
| +AllNodesScan             |
+--------------------------+
----

Internally, however, very different operations need to be executed.
The following table illustrates the difference:

[%header,cols=2*]
|===
|User with `free` role
|User with `restricted` role

|The database can scan all nodes and quickly determine whether nodes are accessible based solely on the nodes' labels (specifically the presence of the `:Customer` label). 

This is a relatively quick operation.

|The database must also access the properties of each of the nodes and inspect their values to make sure the property rule criteria are met (i.e. that `secret` is not set to `true` in this case). 

So due to the additional data access required by the security checks, this operation will be slower compared to executing the query as an unrestricted user.

|===
```

I think using a `MATCH (n) RETURN n` example is more relevant than comparing to accessing the count store.